eslintrc files apply outside their directory

Here's a sample directory structure:

```
/a
  /file.js
/b
  /file.js
  /.eslintrc
/c
  /file.js
file.js
```
- All the javascript files are identical, containing just `var x = "1";`
- `/b/.eslintrc` contains this rule: `quotes: [2, "single"]`

When I run `eslint --reset .` on this directory what I expect to see is an error about the use of double quotes in `/b/file.js`, and nothing else.

What I actually see is:

```
b/file.js
  1:8  error  Strings must use singlequote  quotes

c/file.js
  1:8  error  Strings must use singlequote  quotes

file.js
  1:8  error  Strings must use singlequote  quotes
```

Since there's no warning about `a/file.js`, it appears as though the `b/.eslintrc` file is applying to all files checked after that directory too.

--- a/lib/config.js
+++ b/lib/config.js
@@ -187,7 +187,8 @@ Config.prototype.getConfig = function (filePath) {
         });
     }
 
-    config = this.mergeConfigs(Object.create(this.baseConfig), envConfig);
+    config = this.mergeConfigs({}, this.baseConfig);
+    config = this.mergeConfigs(config, envConfig);
     config = this.mergeConfigs(config, userConfig);
     config = this.mergeConfigs(config, { globals: this.globals });
 

--- a/tests/lib/cli.js
+++ b/tests/lib/cli.js
@@ -285,13 +285,14 @@ describe("cli", function() {
         ];
 
         it("should allow environment-specific globals", function () {
-            cli.execute("--no-eslintrc --config ./conf/eslint.json --env browser,node --no-ignore " + files.join(" "));
-            assert.equal(console.log.args[0][0].split("\n").length, 11);
+            cli.execute("--reset --no-eslintrc --config ./conf/eslint.json --env browser,node --no-ignore " + files.join(" "));
+
+            assert.equal(console.log.args[0][0].split("\n").length, 9);
         });
 
         it("should allow environment-specific globals, with multiple flags", function () {
-            cli.execute("--no-eslintrc --config ./conf/eslint.json --env browser --env node --no-ignore " + files.join(" "));
-            assert.equal(console.log.args[0][0].split("\n").length, 11);
+            cli.execute("--reset --no-eslintrc --config ./conf/eslint.json --env browser --env node --no-ignore " + files.join(" "));
+            assert.equal(console.log.args[0][0].split("\n").length, 9);
         });
     });
 
@@ -302,8 +303,8 @@ describe("cli", function() {
         ];
 
         it("should not define environment-specific globals", function () {
-            cli.execute("--no-eslintrc --config ./conf/eslint.json --no-ignore " + files.join(" "));
-            assert.equal(console.log.args[0][0].split("\n").length, 14);
+            cli.execute("--reset --no-eslintrc --config ./conf/eslint.json --no-ignore " + files.join(" "));
+            assert.equal(console.log.args[0][0].split("\n").length, 12);
         });
     });
 

--- a/tests/lib/config.js
+++ b/tests/lib/config.js
@@ -71,6 +71,22 @@ describe("config", function() {
         });
     });
 
+    describe("getConfig", function() {
+        var firstpath = path.resolve(__dirname, "..", "fixtures", "configurations", "single-quotes", "subdir", ".eslintrc");
+        var secondpath = path.resolve(__dirname, "..", "fixtures", "configurations", "single-quotes", ".eslintrc");
+
+        it("should not retain configs from previous directories", function() {
+            var configHelper = new Config(),
+                config;
+
+            config = configHelper.getConfig(firstpath);
+            assert.equal(0, config.rules["dot-notation"]);
+
+            config = configHelper.getConfig(secondpath);
+            assert.equal(2, config.rules["dot-notation"]);
+        });
+    });
+
     describe("getLocalConfig with directory", function() {
         var code = path.resolve(__dirname, "..", "fixtures", "configurations", "single-quotes", ".eslintrc");
 
@@ -114,14 +130,15 @@ describe("config", function() {
         });
     });
 
-    describe("should cache config", function() {
+    describe("getConfig", function() {
         var code = path.resolve(__dirname, "..", "fixtures", "configurations", "single-quotes", ".eslintrc");
 
-        it("should resolve config only once", function() {
+        it("should cache config", function() {
             var configHelper = new Config();
 
             sinon.spy(configHelper, "findLocalConfigFile");
 
+            // If cached this should be called only once
             configHelper.getConfig(code);
             var callcount = configHelper.findLocalConfigFile.callcount;
             configHelper.getConfig(code);

